### ISSUE TITLE
Memory leak in wazuh-modulesd since 3.2?

### ISSUE THREAD

**Issue by 128432b4117** [wazuh/wazuh#415]
OS: Debian Stretch/stable
Wazuh: 3.2.0-1 installed from packages

Hi,

a few days ago i have updated my wazuh-manager setup from 3.1.0-1 to 3.2.0-1. Since then the memory usage of "wazuh-modulesd" is constantly growing (see screenshot below) until it is using all available memory and the system starts swapping.

Could it be possible that there are some memory leaks introduced in 3.2.0? Is there anything else i can check why the upgraded system is showing this behavior?

Note: The screenshot was done after the restart a few hours ago as the system wasn't usable anymore. Before wazuh-modulesd took around 50% of the available memory (16GB) and around 8 GB of swap was used.

![screenshot_2018-02-24_13-11-56](https://user-images.githubusercontent.com/36791101/36630317-f16e7612-1964-11e8-980b-e29130a023c7.png)

**Comment by 128432b4117** [comment]
Could this be related to the new vulnerability-detector module? I'm currently seeing the following hourly running in my ossec.log:

```
2018/02/24 12:38:54 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Ubuntu Xenial DB update...
2018/02/24 12:39:13 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Ubuntu Trusty DB update...
2018/02/24 12:39:32 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Ubuntu Precise DB update...
2018/02/24 12:39:32 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Red Hat Enterprise Linux 5 DB update...
2018/02/24 12:39:33 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Red Hat Enterprise Linux 6 DB update...
2018/02/24 12:39:33 wazuh-modulesd:vulnerability-detector: INFO: (5461): Starting Red Hat Enterprise Linux 7 DB updat
```

**Comment by 128432b4117** [comment]
Set:

```
  <wodle name="vulnerability-detector">
    <disabled>yes</disabled>
```

in /var/ossec/etc/ossec.conf yesterday. After another 24 hours the memory usage is still low and no excessive memory usage of the wazuh-modulesd is showing up here.

**Comment by crolopez** [comment]
Hi @128432b4117,

These memory leaks of vulnerability-detector wodle will be fixed in Wazuh 3.2.1, which will be released very soon. In the meantime, you can disable that wodle.

Thank you for your feedback and sorry for the inconvenience.

Best regards.

**Comment by 128432b4117** [comment]
Hey, great news. Already updated to Wazuh 3.2.1 and re-enabled the vulnerability-detector wodle again. No high memory usage noticed in the last 24 hours so closing here for now.

Thanks again.
